src/basho_bench.erl

```diff
@@ -346,3 +368,70 @@ get_test_dir() ->
         undefined -> error(unset_test_dir);
         {ok, TestDir} -> TestDir
     end.
+
+master_node() ->
+    case catch basho_bench_config:get(master_node, undefined) of
+        {'EXIT', {noproc, _}} ->
+            case init:get_argument(master_node) of
+                {ok, [[Node]]} -> list_to_atom(Node);
+                _ -> node()
+            end;
+        undefined ->
+            node();
+        Res ->
+            Res
+    end.
+
+node_count() ->
+    case catch basho_bench_config:get(node_count, undefined) of
+        {'EXIT', {noproc, _}} ->
+            case init:get_argument(node_count) of
+                {ok, [[NodeCount]]} -> list_to_integer(NodeCount);
+                _ -> 1
+            end;
+        undefined ->
+            1;
+        Res ->
+            Res
+    end.
+
+is_master() ->
+    case master_node() =:= node() of
+        true -> true;
+        false -> false
+    end.
+
+is_worker() ->
+    not is_master() and is_clustered().
+
+is_clustered() ->
+    case node_count() > 1 of
+        true -> true;
+        false -> false
+    end.
+
+await_nodes(NodeCount) ->
+    await_nodes(NodeCount, 100).
+
+await_nodes(NodeCount, SleepMS) ->
```

Comment:
Perhaps add a timeout in here so we crash after an hour or something?

Reply:
Fixed in 8757911.